### ISSUE TITLE
Python exception when PIV slot does not contain private key - possible due to unsupported keysize

### ISSUE THREAD

**Issue by greenstatic** [Yubico/yubikey-manager#351]
- YubiKey Manager (ykman) version: 3.1.1
- How was it installed?: [Website](https://www.yubico.com/products/services-software/download/yubikey-manager/) and ran the CLI from `/Applications/YubiKey Manager.app/Contents/MacOS/ykman`
- Operating system and version: macOS 10.15.5
- YubiKey model and version: YubiKey 5C, Firmware: 5.1.2
- Bug description summary: Python exception

Steps to reproduce
---

Import a certificate **without private key** into PIV, any slot (I tried 9d, 9e and 82) and run `ykman piv info`. The exception from bellow is triggered.

`yubico-piv-tool -a status` does not have this issue.

Actual results and logs
---
```
Slot 9d:
qml: Return value of PyObject call is NULL: Traceback (most recent call last):

  File "qrc:///py/cli.py", line 12, in run
    return main()

  File "/Applications/YubiKey Manager.app/Contents/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/ykman/cli/__main__.py", line 273, in main
    cli(obj={})

  File "/Applications/YubiKey Manager.app/Contents/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)

  File "/Applications/YubiKey Manager.app/Contents/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)

  File "/Applications/YubiKey Manager.app/Contents/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))

  File "/Applications/YubiKey Manager.app/Contents/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))

  File "/Applications/YubiKey Manager.app/Contents/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)

  File "/Applications/YubiKey Manager.app/Contents/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)

  File "/Applications/YubiKey Manager.app/Contents/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)

  File "/Applications/YubiKey Manager.app/Contents/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/ykman/cli/piv.py", line 182, in info
    algo = ALGO.from_public_key(cert.public_key())

  File "/Applications/YubiKey Manager.app/Contents/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/ykman/piv.py", line 88, in from_public_key
    return getattr(cls, 'RSA%d' % key.key_size)

  File "/Applications/YubiKey Manager.app/Contents/Frameworks/Python.framework/Versions/3.7/lib/python3.7/enum.py", line 349, in __getattr__
    raise AttributeError(name) from None

AttributeError: RSA3072
```

**Comment by dainnilsson** [comment]
No doubt this should be fixed, but out of curiosity, what is the use case for having a certificate without the corresponding private key in a slot?

**Comment by greenstatic** [comment]
My government issues qualified digital certificates to use with government web services. When performing signature verification the entire certificate chain is necessary for successfully validation. The software provided to perform this validation requires that the device contains the entire chain (i.e. intermediate CA, root CA). I solved this by importing the intermediate and root CA into the Retired Key Management slots - although not "proper", it works. It also works if I import it into one of the 4 "user" slots (9a, 9c, 9d and 9e) but then I lose the ability to use them.

**Comment by dainnilsson** [comment]
This is fixed in ykman 4.0.
